database.cc

```diff
@@ -1282,8 +1282,35 @@ std::ostream& operator<<(std::ostream& out, const database& db) {
     return out;
 }
 
-future<mutation> database::do_apply_counter_update(column_family& cf, const frozen_mutation& fm, schema_ptr m_schema,
-                                                   db::timeout_clock::time_point timeout,tracing::trace_state_ptr trace_state) {
+static bool has_deleted_counter_cell(const mutation& state) {
+    bool result = false;
+    for (auto&& cr : state.partition().clustered_rows()) {
+        cr.row().cells().for_each_cell_until([&] (auto id, const atomic_cell_or_collection& ac_o_c) {
+            const auto& cdef = state.schema()->column_at(column_kind::regular_column, id);
+            if (ac_o_c.as_atomic_cell(cdef).is_dead(gc_clock::now())) {
+                result = true;
+                return stop_iteration::yes;
+            }
+            return stop_iteration::no;
+        });
+        if (result) {
+            return result;
+        }
+    }
+    state.partition().static_row().for_each_cell_until([&] (auto id, const atomic_cell_or_collection& ac_o_c) {
+        const auto& cdef = state.schema()->column_at(column_kind::static_column, id);
+        if (ac_o_c.as_atomic_cell(cdef).is_dead(gc_clock::now())) {
+            result = true;
+            return stop_iteration::yes;
+        }
+        return stop_iteration::no;
+    });
+    return result;
```

Comment:
This code repetition should be very easy to deal with.

Reply:
true, will do